src/s3_bench/main.rs

```diff
@@ -0,0 +1,382 @@
+            if (key ^ seed) % 100 < params.prepopulate_percentage {
+                db.put_with_options(
+                    &key.to_be_bytes(),
+                    value.as_slice(),
+                    &WriteOptions {
+                        await_durable: false,
+                    },
+                )
+                .await;
+                inserted += 1;
+            }
+        }
+
+        if !params.plot {
+            println!("Database prepopulated with {} key(s)", inserted);
+        }
+    }
+
+    let counters = (0..params.concurrency)
+        .map(|_| CounterPair {
+            puts: AtomicU64::new(0),
+            gets: AtomicU64::new(0),
+        })
+        .collect();
+    let state = Arc::new(State { db, counters });
+    let mut snapshots: Vec<_> = (0..params.concurrency)
+        .map(|_| SnapshotPair { puts: 0, gets: 0 })
+        .collect();
+
+    let tasks: Vec<_> = (0..params.concurrency)
+        .map(|id| {
+            let params = params.clone();
+            let state = Arc::clone(&state);
+            tokio::spawn(async move {
+                run(id, params, state).await;
+            })
+        })
+        .collect();
+
+    let start = Instant::now();
+    let mut last_sample = start;
+    while start.elapsed() < params.duration {
+        tokio::time::sleep(REPORT_INTERVAL).await;
+
+        let elapsed_since_last_sample = last_sample.elapsed();
+        if elapsed_since_last_sample >= SAMPLE_INTERVAL {
+            last_sample = Instant::now();
+
+            let mut puts = 0u64;
+            let mut gets = 0u64;
+
+            #[allow(clippy::needless_range_loop)]
+            for i in 0..params.concurrency {
+                let v = state.counters[i].puts.load(Ordering::Relaxed);
+                puts += v - snapshots[i].puts;
+                snapshots[i].puts = v;
+
+                let v = state.counters[i].gets.load(Ordering::Relaxed);
+                gets += v - snapshots[i].gets;
+                snapshots[i].gets = v;
+            }
+
+            let elapsed = elapsed_since_last_sample.as_secs_f64();
+            if params.plot {
+                println!(
+                    "{:.3} {:.3}",
+                    // XXX: min is here to align the last sample, so the plot looks nice
+                    f64::min(start.elapsed().as_secs_f64(), params.duration.as_secs_f64()),
+                    (puts + gets) as f64 / elapsed
+                );
+            } else {
+                println!(
+                    "{:.3} put/s + {:.3} get/s = {:.3} op/s, {:.3} s elapsed",
+                    puts as f64 / elapsed,
+                    gets as f64 / elapsed,
+                    (puts + gets) as f64 / elapsed,
+                    elapsed
+                );
+            }
+        }
+    }
+
+    futures::future::join_all(tasks).await;
+
+    state.db.close().await.expect("failed to close database");
+
+    let cache_path = format!("target/s3-bench/cache/s3-bench/{}", db_name);
```

Comment:
Should this be using `root_folder` since the object cache path is configurable?

Reply:
fixed in https://github.com/slatedb/slatedb/pull/233/commits/7bfa9e90974bb4d588f9dbe4ea98fae1679b4aff